### ISSUE TITLE
Exception when using reduce/distinct to aggregate and then mapping the column

### ISSUE THREAD

**Issue by benjamin-asdf** [techascent/tech.ml.dataset#314]
```clojure
(->
 (ds.reduce/group-by-column-agg
  :foo
  {:foos (ds.reduce/distinct :value)}
  (ds/->dataset (into [] (map (fn [i] {:foo 'foo :value (str i)})) (range 3))))
 (ds/column-map :foos-2 (fn [values] values) [:foos]))
```

```#error {
 :cause "Invalid datatype: :iterable"
 :via
 [{:type java.lang.Exception
   :message "Invalid datatype: :iterable"
   :at [tech.v3.datatype.casting$ensure_valid_datatype invokeStatic "casting.clj" 113]}]
 :trace
 [[tech.v3.datatype.casting$ensure_valid_datatype invokeStatic "casting.clj" 113]
  [tech.v3.datatype.casting$ensure_valid_datatype invoke "casting.clj" 110]
  [tech.v3.datatype.base$as_reader invokeStatic "base.clj" 138]
  [tech.v3.datatype.base$as_reader invoke "base.clj" 125]
  [tech.v3.datatype.base$__GT_reader invokeStatic "base.clj" 155]
  [tech.v3.datatype.base$__GT_reader invoke "base.clj" 146]
  [tech.v3.datatype.argops$ensure_reader invokeStatic "argops.clj" 51]
  [tech.v3.datatype.argops$ensure_reader invoke "argops.clj" 38]
  [tech.v3.datatype.emap$emap_reader$fn__20198 invoke "emap.clj" 23]
  [clojure.core$mapv$fn__8535 invoke "core.clj" 6979]
  [clojure.lang.ArrayChunk reduce "ArrayChunk.java" 58]
  [clojure.core.protocols$fn__8244 invokeStatic "protocols.clj" 136]
  [clojure.core.protocols$fn__8244 invoke "protocols.clj" 124]
  [clojure.core.protocols$fn__8204$G__8199__8213 invoke "protocols.clj" 19]
  [clojure.core.protocols$seq_reduce invokeStatic "protocols.clj" 31]
  [clojure.core.protocols$fn__8234 invokeStatic "protocols.clj" 75]
  [clojure.core.protocols$fn__8234 invoke "protocols.clj" 75]
  [clojure.core.protocols$fn__8178$G__8173__8191 invoke "protocols.clj" 13]
  [clojure.core$reduce invokeStatic "core.clj" 6886]
  [clojure.core$mapv invokeStatic "core.clj" 6970]
  [clojure.core$mapv invoke "core.clj" 6970]
  [tech.v3.datatype.emap$emap_reader invokeStatic "emap.clj" 23]
  [tech.v3.datatype.emap$emap_reader invoke "emap.clj" 20]
  [tech.v3.datatype.emap$emap invokeStatic "emap.clj" 161]
  [tech.v3.datatype.emap$emap doInvoke "emap.clj" 124]
  [clojure.lang.RestFn applyTo "RestFn.java" 142]
  [clojure.core$apply invokeStatic "core.clj" 671]
  [clojure.core$apply invoke "core.clj" 662]
  [tech.v3.datatype$emap invokeStatic "datatype.clj" 347]
  [tech.v3.datatype$emap doInvoke "datatype.clj" 337]
  [clojure.lang.RestFn applyTo "RestFn.java" 142]
  [clojure.core$apply invokeStatic "core.clj" 671]
  [clojure.core$apply invoke "core.clj" 662]
  [tech.v3.dataset.column$column_map invokeStatic "column.clj" 231]
  [tech.v3.dataset.column$column_map doInvoke "column.clj" 204]
  [clojure.lang.RestFn applyTo "RestFn.java" 142]
  [clojure.core$apply invokeStatic "core.clj" 671]
  [clojure.core$apply invoke "core.clj" 662]
  [tech.v3.dataset_api$column_map$fn__27230 invoke "dataset_api.clj" 646]
  [clojure.lang.AFn applyToHelper "AFn.java" 154]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 669]
  [clojure.core$apply invoke "core.clj" 662]
  [tech.v3.dataset_api$update invokeStatic "dataset_api.clj" 442]
  [tech.v3.dataset_api$update doInvoke "dataset_api.clj" 418]
  [clojure.lang.RestFn invoke "RestFn.java" 445]
  [tech.v3.dataset_api$column_map invokeStatic "dataset_api.clj" 645]
  [tech.v3.dataset_api$column_map invoke "dataset_api.clj" 540]
  [tech.v3.dataset_api$column_map invokeStatic "dataset_api.clj" 649]
  [tech.v3.dataset_api$column_map invoke "dataset_api.clj" 540]
  [tech.v3.dataset$column_map invokeStatic "dataset.clj" 389]
  [tech.v3.dataset$column_map invoke "dataset.clj" 282]
  [org.sg.prps_analytics.notebooks.high_exporters$eval65499 invokeStatic "NO_SOURCE_FILE" 235]
  [org.sg.prps_analytics.notebooks.high_exporters$eval65499 invoke "NO_SOURCE_FILE" 230]
  [clojure.lang.Compiler eval "Compiler.java" 7194]
  [clojure.lang.Compiler eval "Compiler.java" 7149]
  [clojure.core$eval invokeStatic "core.clj" 3215]
  [clojure.core$eval invoke "core.clj" 3211]
  [nrepl.middleware.interruptible_eval$evaluate$fn__1245$fn__1246 invoke "interruptible_eval.clj" 87]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1990]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1990]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [nrepl.middleware.interruptible_eval$evaluate$fn__1245 invoke "interruptible_eval.clj" 87]
  [clojure.main$repl$read_eval_print__9206$fn__9209 invoke "main.clj" 437]
  [clojure.main$repl$read_eval_print__9206 invoke "main.clj" 437]
  [clojure.main$repl$fn__9215 invoke "main.clj" 458]
  [clojure.main$repl invokeStatic "main.clj" 458]
  [clojure.main$repl doInvoke "main.clj" 368]
  [clojure.lang.RestFn invoke "RestFn.java" 1523]
  [nrepl.middleware.interruptible_eval$evaluate invokeStatic "interruptible_eval.clj" 84]
  [nrepl.middleware.interruptible_eval$evaluate invoke "interruptible_eval.clj" 56]
  [nrepl.middleware.interruptible_eval$interruptible_eval$fn__1278$fn__1282 invoke "interruptible_eval.clj" 152]
  [clojure.lang.AFn run "AFn.java" 22]
  [nrepl.middleware.session$session_exec$main_loop__1348$fn__1352 invoke "session.clj" 218]
  [nrepl.middleware.session$session_exec$main_loop__1348 invoke "session.clj" 217]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.lang.Thread run "Thread.java" 833]]}
```

**Comment by benjamin-asdf** [comment]
```   Invalid datatype: :iterable

               casting.clj:  113  tech.v3.datatype.casting/ensure-valid-datatype
               casting.clj:  110  tech.v3.datatype.casting/ensure-valid-datatype
                  base.clj:  138  tech.v3.datatype.base/as-reader
                  base.clj:  125  tech.v3.datatype.base/as-reader
                  base.clj:  155  tech.v3.datatype.base/->reader
                  base.clj:  146  tech.v3.datatype.base/->reader
                argops.clj:   51  tech.v3.datatype.argops/ensure-reader
                argops.clj:   38  tech.v3.datatype.argops/ensure-reader
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader/fn
                  core.clj: 6979  clojure.core/mapv/fn
             ArraySeq.java:  114  clojure.lang.ArraySeq/reduce
                  core.clj: 6885  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader
                  emap.clj:   20  tech.v3.datatype.emap/emap-reader
                  emap.clj:  161  tech.v3.datatype.emap/emap
                  emap.clj:  124  tech.v3.datatype.emap/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
              datatype.clj:  347  tech.v3.datatype/emap
              datatype.clj:  337  tech.v3.datatype/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
                column.clj:  231  tech.v3.dataset.column/column-map
                column.clj:  204  tech.v3.dataset.column/column-map
               RestFn.java:  442  clojure.lang.RestFn/invoke
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!/fn
                  core.clj: 2586  clojure.core/comp/fn
                  core.clj: 6979  clojure.core/mapv/fn
             protocols.clj:   49  clojure.core.protocols/iter-reduce
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   13  clojure.core.protocols/fn/G
                  core.clj: 6886  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  314  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  381  tech.v3.dataset.io.univocity/eval26555/fn
              MultiFn.java:  239  clojure.lang.MultiFn/invoke
                    io.clj:  277  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
                    io.clj:  279  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
               dataset.clj: 1469  tech.v3.dataset/write!
               dataset.clj: 1445  tech.v3.dataset/write!
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
             Compiler.java: 7194  clojure.lang.Compiler/eval
             Compiler.java: 7149  clojure.lang.Compiler/eval
                  core.clj: 3215  clojure.core/eval
                  core.clj: 3211  clojure.core/eval
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  667  clojure.core/apply
                  core.clj: 1990  clojure.core/with-bindings*
                  core.clj: 1990  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn
                  main.clj:  437  clojure.main/repl/read-eval-print/fn
                  main.clj:  437  clojure.main/repl/read-eval-print
                  main.clj:  458  clojure.main/repl/fn
                  main.clj:  458  clojure.main/repl
                  main.clj:  368  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   84  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   56  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  152  nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
                  AFn.java:   22  clojure.lang.AFn/run
               session.clj:  218  nrepl.middleware.session/session-exec/main-loop/fn
               session.clj:  217  nrepl.middleware.session/session-exec/main-loop
                  AFn.java:   22  clojure.lang.AFn/run
               Thread.java:  833  java.lang.Thread/run

```

**Comment by benjamin-asdf** [comment]
```   Invalid datatype: :iterable

               casting.clj:  113  tech.v3.datatype.casting/ensure-valid-datatype
               casting.clj:  110  tech.v3.datatype.casting/ensure-valid-datatype
                  base.clj:  138  tech.v3.datatype.base/as-reader
                  base.clj:  125  tech.v3.datatype.base/as-reader
                  base.clj:  155  tech.v3.datatype.base/->reader
                  base.clj:  146  tech.v3.datatype.base/->reader
                argops.clj:   51  tech.v3.datatype.argops/ensure-reader
                argops.clj:   38  tech.v3.datatype.argops/ensure-reader
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader/fn
                  core.clj: 6979  clojure.core/mapv/fn
             ArraySeq.java:  114  clojure.lang.ArraySeq/reduce
                  core.clj: 6885  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader
                  emap.clj:   20  tech.v3.datatype.emap/emap-reader
                  emap.clj:  161  tech.v3.datatype.emap/emap
                  emap.clj:  124  tech.v3.datatype.emap/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
              datatype.clj:  347  tech.v3.datatype/emap
              datatype.clj:  337  tech.v3.datatype/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
                column.clj:  231  tech.v3.dataset.column/column-map
                column.clj:  204  tech.v3.dataset.column/column-map
               RestFn.java:  442  clojure.lang.RestFn/invoke
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!/fn
                  core.clj: 2586  clojure.core/comp/fn
                  core.clj: 6979  clojure.core/mapv/fn
             protocols.clj:   49  clojure.core.protocols/iter-reduce
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   13  clojure.core.protocols/fn/G
                  core.clj: 6886  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  314  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  381  tech.v3.dataset.io.univocity/eval26555/fn
              MultiFn.java:  239  clojure.lang.MultiFn/invoke
                    io.clj:  277  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
                    io.clj:  279  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
               dataset.clj: 1469  tech.v3.dataset/write!
               dataset.clj: 1445  tech.v3.dataset/write!
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
             Compiler.java: 7194  clojure.lang.Compiler/eval
             Compiler.java: 7149  clojure.lang.Compiler/eval
                  core.clj: 3215  clojure.core/eval
                  core.clj: 3211  clojure.core/eval
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  667  clojure.core/apply
                  core.clj: 1990  clojure.core/with-bindings*
                  core.clj: 1990  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn
                  main.clj:  437  clojure.main/repl/read-eval-print/fn
                  main.clj:  437  clojure.main/repl/read-eval-print
                  main.clj:  458  clojure.main/repl/fn
                  main.clj:  458  clojure.main/repl
                  main.clj:  368  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   84  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   56  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  152  nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
                  AFn.java:   22  clojure.lang.AFn/run
               session.clj:  218  nrepl.middleware.session/session-exec/main-loop/fn
               session.clj:  217  nrepl.middleware.session/session-exec/main-loop
                  AFn.java:   22  clojure.lang.AFn/run
               Thread.java:  833  java.lang.Thread/run
```

**Comment by benjamin-asdf** [comment]
```   Invalid datatype: :iterable

               casting.clj:  113  tech.v3.datatype.casting/ensure-valid-datatype
               casting.clj:  110  tech.v3.datatype.casting/ensure-valid-datatype
                  base.clj:  138  tech.v3.datatype.base/as-reader
                  base.clj:  125  tech.v3.datatype.base/as-reader
                  base.clj:  155  tech.v3.datatype.base/->reader
                  base.clj:  146  tech.v3.datatype.base/->reader
                argops.clj:   51  tech.v3.datatype.argops/ensure-reader
                argops.clj:   38  tech.v3.datatype.argops/ensure-reader
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader/fn
                  core.clj: 6979  clojure.core/mapv/fn
             ArraySeq.java:  114  clojure.lang.ArraySeq/reduce
                  core.clj: 6885  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader
                  emap.clj:   20  tech.v3.datatype.emap/emap-reader
                  emap.clj:  161  tech.v3.datatype.emap/emap
                  emap.clj:  124  tech.v3.datatype.emap/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
              datatype.clj:  347  tech.v3.datatype/emap
              datatype.clj:  337  tech.v3.datatype/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
                column.clj:  231  tech.v3.dataset.column/column-map
                column.clj:  204  tech.v3.dataset.column/column-map
               RestFn.java:  442  clojure.lang.RestFn/invoke
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!/fn
                  core.clj: 2586  clojure.core/comp/fn
                  core.clj: 6979  clojure.core/mapv/fn
             protocols.clj:   49  clojure.core.protocols/iter-reduce
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   13  clojure.core.protocols/fn/G
                  core.clj: 6886  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  314  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  381  tech.v3.dataset.io.univocity/eval26555/fn
              MultiFn.java:  239  clojure.lang.MultiFn/invoke
                    io.clj:  277  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
                    io.clj:  279  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
               dataset.clj: 1469  tech.v3.dataset/write!
               dataset.clj: 1445  tech.v3.dataset/write!
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
             Compiler.java: 7194  clojure.lang.Compiler/eval
             Compiler.java: 7149  clojure.lang.Compiler/eval
                  core.clj: 3215  clojure.core/eval
                  core.clj: 3211  clojure.core/eval
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  667  clojure.core/apply
                  core.clj: 1990  clojure.core/with-bindings*
                  core.clj: 1990  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn
                  main.clj:  437  clojure.main/repl/read-eval-print/fn
                  main.clj:  437  clojure.main/repl/read-eval-print
                  main.clj:  458  clojure.main/repl/fn
                  main.clj:  458  clojure.main/repl
                  main.clj:  368  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   84  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   56  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  152  nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
                  AFn.java:   22  clojure.lang.AFn/run
               session.clj:  218  nrepl.middleware.session/session-exec/main-loop/fn
               session.clj:  217  nrepl.middleware.session/session-exec/main-loop
                  AFn.java:   22  clojure.lang.AFn/run
               Thread.java:  833  java.lang.Thread/run
```
I think the same issue with I try to write! to a csv.

**Comment by benjamin-asdf** [comment]
```   Invalid datatype: :iterable

               casting.clj:  113  tech.v3.datatype.casting/ensure-valid-datatype
               casting.clj:  110  tech.v3.datatype.casting/ensure-valid-datatype
                  base.clj:  138  tech.v3.datatype.base/as-reader
                  base.clj:  125  tech.v3.datatype.base/as-reader
                  base.clj:  155  tech.v3.datatype.base/->reader
                  base.clj:  146  tech.v3.datatype.base/->reader
                argops.clj:   51  tech.v3.datatype.argops/ensure-reader
                argops.clj:   38  tech.v3.datatype.argops/ensure-reader
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader/fn
                  core.clj: 6979  clojure.core/mapv/fn
             ArraySeq.java:  114  clojure.lang.ArraySeq/reduce
                  core.clj: 6885  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
                  emap.clj:   23  tech.v3.datatype.emap/emap-reader
                  emap.clj:   20  tech.v3.datatype.emap/emap-reader
                  emap.clj:  161  tech.v3.datatype.emap/emap
                  emap.clj:  124  tech.v3.datatype.emap/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
              datatype.clj:  347  tech.v3.datatype/emap
              datatype.clj:  337  tech.v3.datatype/emap
               RestFn.java:  142  clojure.lang.RestFn/applyTo
                  core.clj:  671  clojure.core/apply
                  core.clj:  662  clojure.core/apply
                column.clj:  231  tech.v3.dataset.column/column-map
                column.clj:  204  tech.v3.dataset.column/column-map
               RestFn.java:  442  clojure.lang.RestFn/invoke
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!/fn
                  core.clj: 2586  clojure.core/comp/fn
                  core.clj: 6979  clojure.core/mapv/fn
             protocols.clj:   49  clojure.core.protocols/iter-reduce
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   75  clojure.core.protocols/fn
             protocols.clj:   13  clojure.core.protocols/fn/G
                  core.clj: 6886  clojure.core/reduce
                  core.clj: 6970  clojure.core/mapv
                  core.clj: 6970  clojure.core/mapv
             univocity.clj:  343  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  314  tech.v3.dataset.io.univocity/write-csv!
             univocity.clj:  381  tech.v3.dataset.io.univocity/eval26555/fn
              MultiFn.java:  239  clojure.lang.MultiFn/invoke
                    io.clj:  277  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
                    io.clj:  279  tech.v3.dataset.io/write!
                    io.clj:  252  tech.v3.dataset.io/write!
               dataset.clj: 1469  tech.v3.dataset/write!
               dataset.clj: 1445  tech.v3.dataset/write!
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
                      REPL:  269  org.sg.prps-analytics.notebooks.high-exporters/eval72470
             Compiler.java: 7194  clojure.lang.Compiler/eval
             Compiler.java: 7149  clojure.lang.Compiler/eval
                  core.clj: 3215  clojure.core/eval
                  core.clj: 3211  clojure.core/eval
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  667  clojure.core/apply
                  core.clj: 1990  clojure.core/with-bindings*
                  core.clj: 1990  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  nrepl.middleware.interruptible-eval/evaluate/fn
                  main.clj:  437  clojure.main/repl/read-eval-print/fn
                  main.clj:  437  clojure.main/repl/read-eval-print
                  main.clj:  458  clojure.main/repl/fn
                  main.clj:  458  clojure.main/repl
                  main.clj:  368  clojure.main/repl
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   84  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   56  nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  152  nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
                  AFn.java:   22  clojure.lang.AFn/run
               session.clj:  218  nrepl.middleware.session/session-exec/main-loop/fn
               session.clj:  217  nrepl.middleware.session/session-exec/main-loop
                  AFn.java:   22  clojure.lang.AFn/run
               Thread.java:  833  java.lang.Thread/run
```

I think this is the same bug when writing to csv. Nippy works.

**Comment by benjamin-asdf** [comment]
```clojure
(->
 (ds.reduce/group-by-column-agg
  :foo
  {:foos (ds.reduce/distinct :value)}
  (ds/->dataset (into [] (map (fn [i] {:foo 'foo :value (str i)})) (range 3))))
 (ds/column-map :foos-2 (fn [values] values) [:foos]))
 ```
 can confirm this also throws the err

**Comment by cnuernber** [comment]
Fixed in release 6.092
